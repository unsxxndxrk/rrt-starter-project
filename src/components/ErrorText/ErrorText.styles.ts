import { makeStyles, Theme } from "@material-ui/core/styles";

const useErrorTextStyles = makeStyles((theme: Theme) => ({
  root: {
    color: theme.palette.error.main,
    fontSize: "16px",
    fontWeight: 400,
    padding: "10px 0",
    textAlign: "left",
  },
}));

export { useErrorTextStyles };
