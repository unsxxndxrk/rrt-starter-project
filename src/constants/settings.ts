export const APP_DEFAULT_LANGUAGE: string = "en";
export const API_ENDPOINT: string =
  process.env.REACT_APP_API_ENDPOINT || window.location.origin;
