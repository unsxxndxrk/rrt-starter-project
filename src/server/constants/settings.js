const ORIGIN = "http://localhost";
const PORT = "8000";
const USERS_PATH = "./data/users.json";
const TODOS_PATH = "./data/todos.json";

exports.ORIGIN = ORIGIN;
exports.PORT = PORT;
exports.USERS_PATH = USERS_PATH;
exports.TODOS_PATH = TODOS_PATH;
