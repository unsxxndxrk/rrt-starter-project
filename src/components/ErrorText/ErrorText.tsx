import { useErrorTextStyles } from "./ErrorText.styles";
import cx from "classnames";
import { Namespaces } from "i18n";
import { useTranslation } from "react-i18next";

type ErrorTextProps = {
  text?: string;
  className?: string;
  type?: "unexpected";
  variant?: "tiny";
};

function ErrorText(props: ErrorTextProps) {
  const { text, className, type, variant } = props;
  const classes = useErrorTextStyles();
  const { t } = useTranslation(Namespaces.Common);

  const rootClasses = cx(
    classes.root,
    {
      [classes.tiny]: variant === "tiny",
    },
    className,
  );

  return (
    <div className={rootClasses}>
      {type === "unexpected" ? t("errors.unexpected") : text ? text : null}
    </div>
  );
}

export default ErrorText;
