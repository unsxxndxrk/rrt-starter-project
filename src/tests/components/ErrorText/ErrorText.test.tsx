import { render } from "tests/settings/render";
import ErrorText from "components/ErrorText/ErrorText";

const errorTextMock: string = "error_mock";
const classNameMock: string = "class_name_mock";

describe("<ErrorText />", () => {
  test("should render ErrorText component with custom text", () => {
    const wrapper = render(<ErrorText text={errorTextMock} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
    expect(wrapper.getByText(errorTextMock)).toBeTruthy();
  });

  test("should render ErrorText component with unexpected error", () => {
    const wrapper = render(<ErrorText type="unexpected" />);
    expect(wrapper.asFragment()).toMatchSnapshot();
    expect(wrapper.getByText("errors.unexpected")).toBeTruthy();
  });

  test("should render ErrorText component with custom className", () => {
    const wrapper = render(
      <ErrorText text={errorTextMock} className={classNameMock} />,
    );
    expect(wrapper.asFragment()).toMatchSnapshot();
    expect(wrapper.container.getElementsByClassName(classNameMock).length).toBe(
      1,
    );
  });

  test("should render ErrorText component with variant=tiny", () => {
    const wrapper = render(<ErrorText text={errorTextMock} variant="tiny" />);
    expect(wrapper.asFragment()).toMatchSnapshot();
    expect(
      wrapper.container.getElementsByClassName("makeStyles-tiny").length,
    ).toBe(1);
  });
});
