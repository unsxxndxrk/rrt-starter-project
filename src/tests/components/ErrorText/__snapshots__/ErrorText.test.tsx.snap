// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorText /> should render ErrorText component with custom className 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root class_name_mock"
  >
    error_mock
  </div>
</DocumentFragment>
`;

exports[`<ErrorText /> should render ErrorText component with custom text 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root"
  >
    error_mock
  </div>
</DocumentFragment>
`;

exports[`<ErrorText /> should render ErrorText component with unexpected error 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root"
  >
    errors.unexpected
  </div>
</DocumentFragment>
`;

exports[`<ErrorText /> should render ErrorText component with variant=tiny 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root makeStyles-tiny"
  >
    error_mock
  </div>
</DocumentFragment>
`;
